tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs

```diff
@@ -387,27 +398,31 @@ public void SearchHandler(string query)
     [RelayCommand]
     public void ProviderHandler(int selectedIndex)
     {
-        SelectedProviderIndex = selectedIndex;
-        var currentProvider = Providers[SelectedProviderIndex];
-        ComputeSystemCardsView.Filter = system =>
+        // Swap positions of the selected provider and the first provider in the list
+        // so that the selected provider is always at the top of the list and no extra
+        // UI space is shown.
+        if (selectedIndex != 0)
         {
-            if (currentProvider.Equals(_stringResource.GetLocalized("AllProviders"), StringComparison.OrdinalIgnoreCase))
-            {
-                return true;
-            }
-
-            if (system is CreateComputeSystemOperationViewModel createComputeSystemOperationViewModel)
+            var actualIndex = -1;
+            for (var i = 0; i < PerProviderViewModels.Count; i++)
             {
-                return createComputeSystemOperationViewModel.ProviderDisplayName.Equals(currentProvider, StringComparison.OrdinalIgnoreCase);
+                if (PerProviderViewModels[i].ProviderName.Equals(Providers[selectedIndex], StringComparison.OrdinalIgnoreCase))
+                {
+                    actualIndex = i;
+                    break;
+                }
             }
 
-            if (system is ComputeSystemViewModel computeSystemViewModel)
-            {
-                return computeSystemViewModel.ProviderDisplayName.Equals(currentProvider, StringComparison.OrdinalIgnoreCase);
-            }
+            var temp = PerProviderViewModels[0];
+            PerProviderViewModels[0] = PerProviderViewModels[actualIndex];
+            PerProviderViewModels[actualIndex] = temp;

```

Comment:
Same note about using BulkObservableCollection and BeginBulkOperation/EndBulkOperation. Without that, changing PerProviderViewModels here will fire two UX update events and could cause flickering where we'd show PerProviderViewModels[actualIndex] twice in the list (after assigning to PerProviderViewModels[0] but before the next line).